src/components/MultiHopTrade/components/SharedTradeInput/QuoteSortSelector.tsx

```diff
@@ -0,0 +1,126 @@
+import { Button, ButtonGroup, FormControl, InputGroup } from '@chakra-ui/react'
+import type { FC } from 'react'
+import { memo, useCallback } from 'react'
+import { useTranslate } from 'react-polyglot'
+
+import { HelperTooltip } from '@/components/HelperTooltip/HelperTooltip'
+import { Row } from '@/components/Row/Row'
+import { Text } from '@/components/Text'
+import { selectQuoteSortOption } from '@/state/slices/tradeQuoteSlice/selectors'
+import { tradeQuoteSlice } from '@/state/slices/tradeQuoteSlice/tradeQuoteSlice'
+import { QuoteSortOption } from '@/state/slices/tradeQuoteSlice/types'
+import { useAppDispatch, useAppSelector } from '@/state/store'
+
+enum SortType {
+  BestRate = 'BestRate',
+  LowestGas = 'LowestGas',
+  Fastest = 'Fastest',
+}
+
+export const QuoteSortSelector: FC = memo(() => {
+  const dispatch = useAppDispatch()
+  const translate = useTranslate()
+  const currentSortOption = useAppSelector(selectQuoteSortOption)
+
+  const getSortType = useCallback((option: QuoteSortOption): SortType => {
+    switch (option) {
+      case QuoteSortOption.LOWEST_GAS:
+        return SortType.LowestGas
+      case QuoteSortOption.FASTEST:
+        return SortType.Fastest
+      case QuoteSortOption.BEST_RATE:
+      default:
+        return SortType.BestRate
+    }
+  }, [])
+
+  const currentSortType = getSortType(currentSortOption)
+
+  const handleSortTypeChange = useCallback(
+    (sortType: SortType) => {
+      let sortOption: QuoteSortOption
+      switch (sortType) {
+        case SortType.LowestGas:
+          sortOption = QuoteSortOption.LOWEST_GAS
+          break
+        case SortType.Fastest:
+          sortOption = QuoteSortOption.FASTEST
+          break
+        case SortType.BestRate:
+        default:
+          sortOption = QuoteSortOption.BEST_RATE
+      }
+      dispatch(tradeQuoteSlice.actions.setSortOption(sortOption))
+    },
+    [dispatch],
+  )
+
+  const handleBestRateSortTypeChange = useCallback(
+    () => handleSortTypeChange(SortType.BestRate),
+    [handleSortTypeChange],
+  )
+
+  const handleLowestGasSortTypeChange = useCallback(
+    () => handleSortTypeChange(SortType.LowestGas),
+    [handleSortTypeChange],
+  )
+
+  const handleFastestSortTypeChange = useCallback(
+    () => handleSortTypeChange(SortType.Fastest),
+    [handleSortTypeChange],
+  )
```

Comment:
A bit of overhead but yeah, why not

Reply:
Yeah unfortunately we gotta do this if we want to keep things memoized 😢